src/sharding/Shard.js

```diff
@@ -117,11 +131,20 @@ class Shard extends EventEmitter {
           execArgv: this.execArgv,
         })
         .on('message', this._handleMessage.bind(this))
-        .on('exit', this._exitListener);
+        .on('exit', this._exitListener)
+        .on('disconnect', this._handleDisconnect.bind(this, { parentProcessDisconnect: true }));
     } else if (this.manager.mode === 'worker') {
       this.worker = new Worker(path.resolve(this.manager.file), { workerData: this.env })
         .on('message', this._handleMessage.bind(this))
         .on('exit', this._exitListener);
+
+      const parentPort = require('worker_threads').parentPort;
+      this.parentPort = parentPort;
+      setInterval(() => {
+        if (!parentPort) {
```

Comment:
wouldn't this always be truthy? the variable is never re-assigned as its a const
and should the interval be cleared if this condition is true?